src/sidebar/components/FilterStatus.js

```diff
@@ -49,271 +26,234 @@ import { useRootThread } from './hooks/use-root-thread';
-    return (
-      <FocusFilterStatus filterState={filterState} rootThread={rootThread} />
-    );
-  }
-  return null;
+  // Number of items that match the current filters
+  const resultCount = useMemo(() => {
+    return filterMode === 'selection'
+      ? selectedCount
+      : countVisible(rootThread) - forcedVisibleCount;
+  }, [filterMode, selectedCount, rootThread, forcedVisibleCount]);
+
+  // Number of additional items that are visible but do not match current
+  // filters. This can happen when, e.g.:
+  //  - A user manually expands a thread that does not match the current
+  //    filtering
+  //  - A user creates a new annotation when there are applied filters
+  const additionalCount = useMemo(() => {
+    if (filterMode === 'selection') {
+      // Selection filtering deals in top-level annotations only.
+      // Compare visible top-level annotations against the count of selected
+      // (top-level) annotatinos.
+      const visibleAnnotationCount = (rootThread.children || []).filter(
+        thread => thread.annotation && thread.visible
+      ).length;
+      return visibleAnnotationCount - selectedCount;
+    } else {
+      return forcedVisibleCount;
+    }
+  }, [filterMode, forcedVisibleCount, rootThread.children, selectedCount]);
+
+  const buttonText = useMemo(() => {
+    if (filterMode === 'selection') {
+      // Because of the confusion between counts of entities between selected
+      // annotations and filtered annotations, don't display the total number
+      // when in user-focus mode because the numbers won't appear to make sense.
+      // Don't display total count, either, when viewing a direct-linked annotation.
+      const showCount = !focusState.configured && !directLinkedId;
+      return showCount ? `Show all (${annotationCount})` : 'Show all';
+    } else if (filterMode === 'focus') {
+      if (forcedVisibleCount > 0) {
+        return 'Reset filters';
+      }
+      return focusState.active
+        ? 'Show all'
+        : `Show only ${focusState.displayName}`;
+    }
+    return 'Clear search';
+  }, [
+    annotationCount,
+    directLinkedId,
+    focusState,
+    filterMode,
+    forcedVisibleCount,
+  ]);
+
+  return (
+    <Card
+      classes={classNames('mb-3 p-3', {
+        // This container element needs to be present at all times but
+        // should only be visible when there are applied filters
+        'sr-only': !filterMode,
+      })}
+    >
+      <div className="flex items-center justify-center space-x-1">
+        {store.isLoading() ? (
+          <Spinner />
+        ) : (
+          <>
+            <div
+              className={classNames(
+                // Setting `min-width: 0` here allows wrapping to work as
+                // expected for long `filterQuery` strings. See
+                // https://css-tricks.com/flexbox-truncated-text/
+                'grow min-w-[0]'
+              )}
+              role="status"
+            >
+              <>
```

Comment:
This `<>` fragment appears to be unnecessary, since there is only one child expression and we're already in an element children context where a sequence of multiple expressions can appear without being wrapped.